File: 4348-maximum-total-sum-of-k-selected-elements/maximum-total-sum-of-k-selected-elements.java
class Solution {
    public long maxSum(int[] nums, int k, int mul) {
        Arrays.sort(nums);
        int n = nums.length;

        long ans = 0;
        for(int i = n-1; i>n-k-1; i--){
            // System.out.println(i);
            if(mul >0){
                ans  = ans + (long)(nums[i]) * mul;
                mul--;
            }
            else{
                ans+=nums[i];
            }
        }
        return ans;
    }
}

File: 4348-maximum-total-sum-of-k-selected-elements/maximum-total-sum-of-k-selected-elements.py
class Solution:
    def maxSum(self, nums: list[int], k: int, mul: int) -> int:
        ans = 0
        nums.sort(reverse = True)
        for i in range(0, k, 1):
            if(mul > 0):
                ans = ans + (nums[i] * mul)
                mul = mul - 1
            else:
                ans = ans + nums[i]
        return ans
        